src/wallet/rpcwallet.cpp

```diff
@@ -4232,6 +4256,244 @@ UniValue z_gettotalbalance(const UniValue& params, bool fHelp)
     return result;
 }
 
+UniValue z_getbalances(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() > 2)
+        throw runtime_error(
+            "z_getbalances ( minconf )\n"
+            "\nReturns the balances available for each independent spending authority held by the wallet,\n"
+            "and optionally the balances and received amounts associated with imported watch-only addresses\n"
+            "and viewing keys.\n"
+            "This includes the legacy transparent pool of funds, each distinct Sprout and Sapling address\n"
+            "derived using `z_getnewaddress` or from spending keys imported with `z_importkey`, and funds\n"
+            "held by each HD-derived Unified Account in the wallet.\n"
+            "\nArguments:\n"
+            "1. minconf          (numeric, optional, default=1) Only include private and transparent transactions confirmed at least this many times.\n"
+            "2. includeWatchonly (bool, optional, default=false) Also include balance in watchonly addresses (see 'importaddress' and 'z_importviewingkey')\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"legacy_transparent\": {                (object, optional) the balances of transparent funds held by legacy transparent keys,\n"
```

Comment:
TODO: separate out transparent coinbase value, and display both mature and immature transparent coinbase balances. Also do the same for transparent account value.